tools/cli/commands/generate.js

```diff
@@ -232,26 +236,107 @@ export function getQuestions( type ) {
 export function generatePackage(
 	answers = { name: 'test', description: 'n/a', buildScripts: [] }
 ) {
+	const project = 'packages/' + answers.name;
 	const pkgDir = path.join( __dirname, '../../..', 'projects/packages', answers.name );
+
+	createSkeleton( pluralize( answers.type ), pkgDir, answers.name );
+
+	// Generate the package.json file
+	const packageJson = readPackageJson( project );
+	createPackageJson( packageJson, answers );
+	writePackageJson( project, packageJson, pkgDir );
+
+	// Generate the composer.json file
+	const composerJson = readComposerJson( project );
+	createComposerJson( composerJson, answers );
+	writeComposerJson( project, composerJson, pkgDir );
+
+	return packageJson;
+}
+
+/**
+ * Generate a plugin based on questions passed to it.
+ *
+ * @todo REMOVE EXPORT. ONLY FOR TESTING.
+ *
+ * @param {object} answers - Answers from questions.
+ *
+ * @returns {object} package.json object. TEMPORARY FOR TESTING.
+ */
+export function generatePlugin( answers = { name: 'test', description: 'n/a', buildScripts: [] } ) {
+	const project = 'plugins/' + answers.name;
+	const pluginDir = path.join( __dirname, '../../..', 'projects/plugins', answers.name );
+
+	// Copy the skeletons over
+	createSkeleton( pluralize( answers.type ), pluginDir, answers.name );
+
+	// Generate the package.json file
+	const packageJson = readPackageJson( project );
+	createPackageJson( packageJson, answers );
+	writePackageJson( project, packageJson, pluginDir );
+
+	// Generate the composer.json file
+	const composerJson = readComposerJson( project );
+	createComposerJson( composerJson, answers );
```

Comment:
what does `create` functions do vs `write` functions?

Reply:
nevmind. I'm on board.

Reply:
Just closing the loop here that `create` generates the actual JSON file with our custom content - modified by `createComposerJson` - the `write` function then actually writes the content to the file.

[Source you shared on passing objects as references](https://stackoverflow.com/questions/13104494/does-javascript-pass-by-reference).